teachertool/src/components/DragAndDropFileSurface.tsx

```diff
@@ -0,0 +1,103 @@
+import { classList } from "react-common/components/util";
+import { Strings } from "../constants";
+import { NoticeLabel } from "./NoticeLabel";
+import { useRef, useState } from "react";
+import css from "./styling/DragAndDropFileSurface.module.scss";
+import { Button } from "react-common/components/controls/Button";
+
+export interface DragAndDropFileSurfaceProps {
+    onFileDroppedAsync: (file: File) => void;
+    errorMessage?: string;
+}
+export const DragAndDropFileSurface: React.FC<DragAndDropFileSurfaceProps> = ({ onFileDroppedAsync, errorMessage }) => {
+    const [fileIsOverSurface, setFileIsOverSurface] = useState(false);
+    const [errorKey, setErrorKey] = useState(0); // Used to reset animations on the error label.
```

Comment:
By changing the key of the component, we force react to unmount the component and mount a new one, which resets the animation state. If we don't do this, then the animation only runs on the first error and no subsequent ones, since from the component's perspective, the animation has already run.

Reply:
Ohh, gotcha. Makes sense, so errorKey could technically be any type, as long as we can make a quick change to it to cause an unmount and mount, and that's why using numbers is convenient?

Reply:
As long as it's a valid key, then yes (though I think that's really only strings and numbers).